python/pfs/instmodel/splinedPsf.py

```diff
@@ -972,25 +972,35 @@ def makeDetectorMap(self, fname, obsdate=None):
         dlam = self.spotsInfo['LAMINC']
         lams = np.linspace(lam0, lam0+(nKnots-1)*dlam, nKnots, dtype=np.float32)
 
-        allYKnots = []
-        allXCenters = []
+        xCenterKnots = []
+        xCenterValues = []
+        wavelengthKnots = []
+        wavelengthValues = []
         for holeId in fiberIds:
             coeffs = self.getCoeffs(holeId)
 
             yKnot = self._focalPlaneYToDetectorY(coeffs.ycCoeffs.get_coeffs(), correctToLL=True) / pixelScale
             xc = self._focalPlaneXToDetectorX(coeffs.xcCoeffs.get_coeffs(), correctToLL=True) / pixelScale
-            allYKnots.append(yKnot.astype(np.float32))
-            allXCenters.append(xc.astype(np.float32))
-
             assert len(lams) == len(yKnot)
             assert len(lams) == len(xc)
 
+            xCenterKnots.append(yKnot.astype(np.float32))
```

Comment:
The ``xCenterKnots`` are constructed from ``yKnots``??

Reply:
Yes: the knots from one spline are used in another.

As for the style violations, I think that's falls under a cleanup ticket for the simulator that's never been scheduled.